### PR TITLE
Speed up DIALS .mtz export

### PR DESCRIPTION
**Pull request by Anthchirp**
Instead of running O(n) calls to _add_batch(), which runs an O(n) check
and then runs another C library function running another O(n) check for
a total complexity of O(2n²)...

...run one custom function that inserts all batch information at the
same time for O(n) cost in time.

Relevant ticket: https://github.com/dials/dials/issues/924

### CODE REVIEW

File: iotbx/mtz/object.cpp [hunk 1]
```diff
@@ -246,6 +247,166 @@ namespace iotbx { namespace mtz {
     return result;
   }
 
+  void
+  object::add_dials_batches(int dataset_id,
```
Comment (rjgildea) [inline iotbx/mtz/object.cpp]: Does the name _really_ need to have "dials" in?

Comment (Anthchirp) [reply iotbx/mtz/object.cpp]: We thought this to be sensible as this function takes away a lot of the options for setting values and replaces them with DIALS-specific defaults.

Comment (Anthchirp) [reply iotbx/mtz/object.cpp]: Also - if DIALS ever needs to write different things we'd want to be free to modify the function signature, so really don't want anyone else to rely on this function.

**Comment by Anthchirp** [comment]
@rjgildea right now looking at moving the function into the dials repository instead (which would be a more appropriate place indeed)

File: iotbx/mtz/object.cpp (hunk 1 above)
Comment (Anthchirp) [reply iotbx/mtz/object.cpp]: cf https://github.com/dials/dials/pull/936

**Pull request closed without merging**